cmake_modules/IcebergThirdpartyToolchain.cmake

```diff
@@ -82,13 +84,15 @@ function(resolve_arrow_dependency)
       ON
       CACHE BOOL "" FORCE)
   set(ARROW_DEPENDENCY_SOURCE
-      "AUTO"
+      "BUNDLED"
       CACHE STRING "" FORCE)
 
   fetchcontent_declare(Arrow
                        ${FC_DECLARE_COMMON_OPTIONS}
-                       URL ${ARROW_SOURCE_URL}
-                       URL_HASH "SHA256=${ICEBERG_ARROW_BUILD_SHA256_CHECKSUM}"
+                       GIT_REPOSITORY https://github.com/wgtmac/arrow.git
```

Comment:
This is still on my fork: https://github.com/apache/arrow/pull/46232. Will update it once merged and released.